final class com.seventeenmiles.sketch.cloud.d extends java.util.concurrent.FutureTask
{
    final com.seventeenmiles.sketch.cloud.a a;

    void <init>(com.seventeenmiles.sketch.cloud.a, java.util.concurrent.Callable)
    {
        com.seventeenmiles.sketch.cloud.d r0;
        java.util.concurrent.Callable $r2;
        com.seventeenmiles.sketch.cloud.a $r1;

        r0 := @this: com.seventeenmiles.sketch.cloud.d;

        $r1 := @parameter0: com.seventeenmiles.sketch.cloud.a;

        $r2 := @parameter1: java.util.concurrent.Callable;

        r0.<com.seventeenmiles.sketch.cloud.d: com.seventeenmiles.sketch.cloud.a a> = $r1;

        specialinvoke r0.<java.util.concurrent.FutureTask: void <init>(java.util.concurrent.Callable)>($r2);

        return;
    }

    protected final void done()
    {
        com.seventeenmiles.sketch.cloud.d r0;
        java.lang.Throwable $r3, $r4, $r5;
        com.seventeenmiles.sketch.cloud.a $r1;
        java.lang.RuntimeException $r6;
        java.util.concurrent.ExecutionException $r7;
        java.lang.Object $r2;

        r0 := @this: com.seventeenmiles.sketch.cloud.d;

        $r1 = r0.<com.seventeenmiles.sketch.cloud.d: com.seventeenmiles.sketch.cloud.a a>;

     label1:
        $r2 = virtualinvoke r0.<com.seventeenmiles.sketch.cloud.d: java.lang.Object get()>();

        staticinvoke <com.seventeenmiles.sketch.cloud.a: void c(com.seventeenmiles.sketch.cloud.a,java.lang.Object)>($r1, $r2);

     label2:
        return;

     label3:
        $r3 := @caughtexception;

        staticinvoke <android.util.Log: int w(java.lang.String,java.lang.Throwable)>("AsyncTask", $r3);

        return;

     label4:
        $r4 := @caughtexception;

        $r6 = new java.lang.RuntimeException;

        $r7 = (java.util.concurrent.ExecutionException) $r4;

        $r3 = virtualinvoke $r7.<java.util.concurrent.ExecutionException: java.lang.Throwable getCause()>();

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("An error occured while executing doInBackground()", $r3);

        throw $r6;

     label5:
        $r5 := @caughtexception;

        $r1 = r0.<com.seventeenmiles.sketch.cloud.d: com.seventeenmiles.sketch.cloud.a a>;

        staticinvoke <com.seventeenmiles.sketch.cloud.a: void c(com.seventeenmiles.sketch.cloud.a,java.lang.Object)>($r1, null);

        return;

        catch java.lang.InterruptedException from label1 to label2 with label3;
        catch java.util.concurrent.ExecutionException from label1 to label2 with label4;
        catch java.util.concurrent.CancellationException from label1 to label2 with label5;
    }
}
